libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -530,15 +535,23 @@ public void checkWriteProperty(Class<?> callerClass, String property) {
                 requestingClass,
                 property
             ),
-            callerClass
+            callerClass,
+            entitlements.componentName()
         );
     }
 
-    private void notEntitled(String message, Class<?> callerClass) {
+    private void notEntitled(String message, Class<?> callerClass, String componentName) {
         var exception = new NotEntitledException(message);
         // Don't emit a log for muted classes, e.g. classes containing self tests
         if (mutedClasses.contains(callerClass) == false) {
-            logger.warn(message, exception);
+            var moduleName = callerClass.getModule().getName();
+            var loggerSuffix = "." + componentName + "." + ((moduleName == null) ? ALL_UNNAMED : moduleName);
+            var notEntitledLogger = LogManager.getLogger(PolicyManager.class.getName() + loggerSuffix);
+            String frameInfoSuffix = StackWalker.getInstance(RETAIN_CLASS_REFERENCE)
+                .walk(this::findRequestingFrame)
+                .map(StackFrame::toString)
+                .orElse("");
+            notEntitledLogger.warn(message + frameInfoSuffix);
```

Comment:
Should there be a newline (with indentation) so it's sort of like a stack trace? What does one of these messages look like?

Reply:
The issue we were resolving here was that it looked too much like an error, so I figured making it look like an exception stack trace would be a bit counterproductive. Also, we've experienced situations lately where multi-line logs get split up into multiple logs, which is really inconvenient.

But I don't feel too strongly about it.

Reply:
Here is an example:

```
2025-03-11T12:41:55,308][WARN ][o.e.e.r.p.P.(.o.e.server ] [runTask-0] Not entitled: component [(server)], module [org.elasticsearch.server], class [class org.elasticsearch.bootstrap.Elasticsearch$EntitlementSelfTester], operation [start process]org.elasticsearch.server@9.1.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch$EntitlementSelfTester.ensureCannotStartProcess(Elasticsearch.java:296)
```

In IntelliJ, the file name and line number are a hyperlink.

Reply:
huh I don't like the lack of spacing there tho

Reply:
FWIW, with a newline, a tab, and an `at`, it can look like this:

```
[2025-03-11T12:51:03,821][WARN ][o.e.e.r.p.P.(.o.e.server ] [runTask-0] Not entitled: component [(server)], module [org.elasticsearch.server], class [class org.elasticsearch.bootstrap.Elasticsearch$EntitlementSelfTester], operation [start process]
	at org.elasticsearch.server@9.1.0-SNAPSHOT/org.elasticsearch.bootstrap.Elasticsearch$EntitlementSelfTester.ensureCannotStartProcess(Elasticsearch.java:296)
```